src/transformers/models/gpt2/modeling_gpt2.py

```diff
@@ -1410,7 +1410,7 @@ def forward(
                     "unexpected if using padding tokens in conjunction with `inputs_embeds.`"
                 )
 
-        pooled_logits = logits[torch.arange(batch_size, device=self.device), sequence_lengths]
+        pooled_logits = logits[torch.arange(batch_size, device=logits.device), sequence_lengths]
```

Comment:
Same reason. 
This should not break things because the tensor should be on the same device as `logits` anyway, right?